api/src/main/java/org/openmrs/Diagnosis.java

```diff
@@ -0,0 +1,122 @@
+package org.openmrs;
+
+public class Diagnosis extends BaseOpenmrsData{
+	
+	private Integer diagnosisId;
+	
+	private Encounter encounter;
+	
+	private CodedOrFreeText diagnosis;
+
+	/**
+	 * Condition 
+	 * 
+	 * May refer to a new condition to be created as a result of this diagnosis,
+	 * or an existing condition that this encounter is following up.
+	 * **/
+	private Condition condition;
+
+	private ConditionVerificationStatus certainty;
+
+	/** default constructor */
+	public Diagnosis() {
+	}
+
+	/**
+	 * @param diagnosisId
+	 * @should set diagnosis id
+	 */
+	public Diagnosis(Integer diagnosisId) {
+		this.diagnosisId = diagnosisId;
+	}
+
+	/**
+	 * @return Returns the diagnosisId.
+	 */
+	@Override
+	public Integer getId() {
+		return diagnosisId;
+	}
+
+	/**
+	 * @param id
+	 * @should set diagnosis id
+	 */
+	@Override
+	public void setId(Integer id) {
+		this.diagnosisId = id;
+	}
+
+	/**
+	 * @return Returns the encounter.
+	 */
+	public Encounter getEncounter() {
+		return encounter;
+	}
+
+	/**
```

Comment:
Can you do proper javadoc documentation of these methods? You can take a look at the comments i put in Arthur's pull request.

Reply:
Noted. Looking at it